api/opencl_platform_layer.asciidoc

```diff
@@ -239,34 +239,40 @@ include::{generated}/api/version-notes/CL_DEVICE_TYPE_ACCELERATOR.asciidoc[]
 
 include::{generated}/api/version-notes/CL_DEVICE_TYPE_CUSTOM.asciidoc[]
   | Specialized devices that implement some of the OpenCL runtime APIs but
-    do not support all required OpenCL functionality.
+    do not support all of the required OpenCL functionality.
+    If an OpenCL device reports {CL_DEVICE_TYPE_CUSTOM} in its {CL_DEVICE_TYPE}
+    then it cannot be combined with any other device types.
 
 | {CL_DEVICE_TYPE_DEFAULT_anchor}
 
 include::{generated}/api/version-notes/CL_DEVICE_TYPE_DEFAULT.asciidoc[]
   | The default OpenCL device in the platform.
+    One device in the platform must be returned as the {CL_DEVICE_TYPE_DEFAULT}
+    device when passed as the _device_type_ to {clGetDeviceIDs}, however
+    {CL_DEVICE_TYPE_DEFAULT} must not be reported in {CL_DEVICE_TYPE} for any
+    OpenCL device.
     The default OpenCL device must not be a {CL_DEVICE_TYPE_CUSTOM} device.
 
 | {CL_DEVICE_TYPE_ALL_anchor}
 
 include::{generated}/api/version-notes/CL_DEVICE_TYPE_ALL.asciidoc[]
   | All OpenCL devices available in the platform, except for
     {CL_DEVICE_TYPE_CUSTOM} devices.
+    Only used to query OpenCL devices using {clGetDeviceIDs} and will never be
+    returned as the {CL_DEVICE_TYPE} for any OpenCL device.
 
 |====
 
 The device type is purely informational and has no semantic meaning.
 
-Some devices may be more than one type.
-For example, a {CL_DEVICE_TYPE_CPU} device may also be a
-{CL_DEVICE_TYPE_GPU} device, or a {CL_DEVICE_TYPE_ACCELERATOR} device
-may also be some other, more descriptive device type.
-{CL_DEVICE_TYPE_CUSTOM} devices must not be combined with any other
+[NOTE]
+====
+Although the standard allows an OpenCL device to report more than one device
+type in its {CL_DEVICE_TYPE}, many applications assume that devices will only
+report a single device type. Implementations are strongly encouraged to report a
+single device type for {CL_DEVICE_TYPE} and not to report a combination of
```

Comment:
Text above says `{CL_DEVICE_TYPE_DEFAULT} **must not** be reported in {CL_DEVICE_TYPE}`, but the wording of this note is just **strongly encouraged**.

I'm not sure if any devices have reported more than one non-default device here, e.g. `GPU | ACCELERATOR`, which thing wording discourages but doesn't outright forbid.

Reply:
I strengthened the requirement further and now the device must report only a single device type.